docs/source/conf.py

```diff
@@ -288,16 +287,39 @@
 # further.  For a list of options available for each theme, see the
 # documentation.
 #
+
 html_theme_options = {
     "show_toc_level": 2,
     "use_edit_page_button": True,
+    "logo": {
+      "image_light": "_static/arrow.png",
+      "image_dark": "_static/arrow-dark.png",
+    },
+    "header_links_before_dropdown": 2,
+    "header_dropdown_text": "Language implementations",
+    "navbar_end": ["version-switcher", "theme-switcher", "navbar-icon-links"],
+    "icon_links": [
+        {
+            "name": "GitHub",
+            "url": "https://github.com/apache/arrow",
+            "icon": "fa-brands fa-square-github",
+        },
+        {
+            "name": "Twitter",
+            "url": "https://twitter.com/ApacheArrow",
+            "icon": "fa-brands fa-square-twitter",
+        },
+    ],
+    "show_version_warning_banner": True,
+    "switcher": {
+        # TODO: update versions.json on the arrow-site and change the link here
+        # "json_url": "/docs/_static/versions.json"
+        "json_url": "https://gist.githubusercontent.com/AlenkaF/9cc5820d3030cefcd2eee48e1ea58dc5/raw/3c0fa34dae3926f05958a5a7b3cfa8a07f0846d0/versions.json",
```

Comment:
This needs to be updated before merging this PR?

Reply:
That would be best, yes. Can open a PR on arrow-site.

Reply:
https://github.com/apache/arrow-site/pull/407